cmd/ipfs/main.go

```diff
@@ -671,6 +673,15 @@ func apiClientForAddr(addr ma.Multiaddr) (cmdsHttp.Client, error) {
 }
 
 func isConnRefused(err error) bool {
-	return strings.Contains(err.Error(), "connection refused") ||
-		strings.Contains(err.Error(), "target machine actively refused it")
+	urlerr, ok := err.(*url.Error)
+	if !ok {
+		return false
+	}
+
+	netoperr, ok := urlerr.Err.(*net.OpError)
+	if !ok {
+		return false
+	}
+
+	return netoperr.Op == "dial"
```

Comment:
im not sure this one statement captures the two other errors. also no longer works with a non-url `net.OpError`. and if a dial fails for some other reason (like memory alloc failure or some other random happenstance) this will not discern (may be ok). 

how about something like:

```go
func isConnRefused(err error) bool {
  switch e := err.(type) {
  case *url.Error:
    return isConnRefused(e.Err)
  case *net.OpError:
    return e.Op == "dial" && (further condition that e really means failure to connect here)
  default:
    return false
  }
}
```